nonsemantic/NonSemantic.ClspvReflection.asciidoc

```diff
@@ -677,6 +677,243 @@ specialization id for the value. +
+ +
+Declares a pointer argument for 'Kernel' passed in a *Uniform* buffer.
+The Vulkan descriptor type should use *VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER*. This
+argument may share a descriptor set and binding with other users. +
+ +
+'Kernel' must be a <<Kernel,*Kernel*>> extended instruction from the same import. +
+ +
+'Ordinal' must be a 32-bit unsigned integer *OpConstant* specifying the argument ordinal using
+zero-based counting. +
+ +
+'DescriptorSet' must be a 32-bit unsigned integer *OpConstant* specifying the descriptor set. +
+ +
+'Binding' must be a 32-bit unsigned integer *OpConstant* specifying the binding. +
+ +
+'Offset' must be a 32-bit unsigned integer *OpConstant* specifying the offset in the block in bytes. +
+ +
+'Size' must be a 32-bit unsigned integer *OpConstant* specifying the size of the pointer in bytes. +
+ +
+'ArgInfo' must be an <<ArgumentInfo,*ArgumentInfo*>> extended instruction from the same import.
+1+| 27 | '<id>' +
+'Kernel' |
+'<id>' +
+'Ordinal' |
+'<id>' +
+'DescriptorSet' |
+'<id>' +
+'Binding' |
+'<id>' +
+'Offset' |
+'<id>' +
+'Size' |
+Optional +
+'<id>' +
+'ArgInfo' |
+|=====
+
+[cols="4"]
+|=====
+4+| [[ProgramScopeVariablesStorageBuffer]]*ProgramScopeVariablesStorageBuffer* +
+Declares a storage buffer to hold program scope variables. All kernels from this module
+should include a descriptor with the type *VK_DESCRIPTOR_TYPE_STORAGE_BUFFER* that is backed by
+a buffer initialized with 'Data' and shared between all instances of kernels created from
+this module. +
+ +
+'DescriptorSet' must be a 32-bit unsigned integer *OpConstant* specifying the descriptor set. +
+ +
+'Binding' must be a 32-bit unsigned integer *OpConstant* specifying the binding. +
+ +
+'Data' must be an *OpString* that encodes the hexbytes of the data used to initialize the buffer.
+1+| 28 | '<id>' +
+'DescriptorSet' |
+'<id>' +
+'Binding' |
+'<id>' +
+'Data'
+|=====
+
+[cols="4"]
+|=====
+4+| [[ProgramScopeVariablePointerRelocation]]*ProgramScopeVariablePointerRelocation* +
+Declares a relocation for a pointer into the program scope variables storage buffer
+initialized with the address of another program scope variable. +
+ +
+'ObjectOffset' must be a 32-bit unsigned integer *OpConstant* specifying the offset into the program
+scope variable storage buffer at which the object pointed to resides. +
+ +
+'PointerOffset' must be a 32-bit unsigned integer *OpConstant* specifying the offset into the program
+scope variable storage buffer at which the pointer resides. +
+ +
+'PointerSize' must be a 32-bit unsigned integer *OpConstant* specifying the size of the pointer
+stored in the program scope variable storage buffer.
+1+| 29 | '<id>' +
+'ObjectOffset' |
+'<id>' +
+'PointerOffset' |
+'<id>' +
+'PointerSize'
+|=====
+
+[cols="5"]
+|=====
+5+|[[ImageArgumentInfoChannelOrderPushConstant]]*ImageArgumentInfoChannelOrderPushConstant* +
+ +
+Declares a *PushConstant* location to pass the channel order of the image that
```

Comment:
I think maybe there should be a link for channel order and format entries, but I'm not sure to which spec (probably 5.3.1.1 of the CL spec for channel order).

Reply:
Why not. I've linked the OpenCL C spec since the values passed around here will be the CLK ones rather than the API-side definitions but if you prefer linking the API spec that does have more background on image formats instead, I'm equally fine with that.